### PR TITLE
[Utils] add offensive language classifier util

### PR DESCRIPTION
**Pull request by emilydinan**
**Patch description**
Add a utility module for detecting offensive language.
Rename the original Offensive Language Detector to better convey the difference between the two modules.
Add a test for both the string matching and classifier modules.

### CODE REVIEW

File: tests/test_safety_modules.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# LICENSE file in the root directory of this source tree.
+
+from parlai.utils.safety import OffensiveLanguageClassifier as olc
+from parlai.utils.safety import OffensiveStringMatcher as osm
+from parlai.utils.testing import skipUnlessBPE
+
+
+import unittest
+
+definitely_good = [
```
Comment (stephenroller) [inline tests/test_safety_modules.py]: uppercase constants

File: tests/test_safety_modules.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from parlai.utils.safety import OffensiveLanguageClassifier as olc
```
Comment (stephenroller) [inline tests/test_safety_modules.py]: nit: dont' shorten

File: parlai/utils/testing.py [hunk 3]
```diff
@@ -35,6 +35,13 @@
     git_ = None
     GIT_AVAILABLE = False
 
+try:
+    from subword_nmt import learn_bpe, apply_bpe  # noqa: F401
```
Comment (stephenroller) [inline parlai/utils/testing.py]: nit: `import subword_nmt  # noqa: F401`

File: parlai/utils/testing.py [hunk 4]
```diff
@@ -63,6 +70,11 @@ def skipUnlessGPU(testfn, reason='Test requires a GPU'):
     return unittest.skipUnless(GPU_AVAILABLE, reason)(testfn)
 
 
+def skipUnlessBPE(testfn, reason='Test requires a GPU'):
```
Comment (stephenroller) [inline parlai/utils/testing.py]: Maybe we should just add subword-nmt to the requirements? We can revisit later.

File: parlai/utils/safety.py [hunk 5]
```diff
@@ -5,11 +5,56 @@
...
+        act = {'text': text, 'episode_done': True}
+        self.model.observe(act)
+        response = self.model.act()['text']
+        pred_class, prob = [x.split(': ')[-1] for x in response.split('\n')]
+        pred_not_ok = self.classes[pred_class]  # check whether classified as NOT OK
+        prob = float(prob)  # cast string to float
+
+        return pred_not_ok, prob
+
+    def __contains__(self, key):
```
Comment (stephenroller) [inline parlai/utils/safety.py]: lol please document this weirdness in the class level docstring.

File: parlai/utils/safety.py [hunk 6]
```diff
@@ -5,11 +5,56 @@
...
+    TransformerClassifierAgent,
+)
+from parlai.core.agents import create_agent
+from parlai.tasks.dialogue_safety.agents import OK_CLASS, NOT_OK_CLASS
 
 import os
 
 
-class OffensiveLanguageDetector(object):
+class OffensiveLanguageClassifier:
```
Comment (stephenroller) [inline parlai/utils/safety.py]: Is there a better name that signals how sophisticated this model is?

File: parlai/utils/safety.py (hunk 5 above)
Comment (emilydinan) [reply parlai/utils/safety.py]: 😠

Comment (stephenroller) [reply parlai/utils/safety.py]: 🧟‍♂️

**Pull request merged**